arch/arc/core/cache.c

```diff
@@ -117,29 +100,38 @@ static void dcache_flush_mlines(u32_t start_addr, u32_t size)
 
 }
 
-
-/**
- *
- * @brief Flush d-cache lines to main memory
- *
- * No alignment is required for either <virt> or <size>, but since
- * sys_cache_flush() iterates on the d-cache lines, a d-cache line alignment for
- * both is optimal.
- *
- * The d-cache line size is specified either via the CONFIG_CACHE_LINE_SIZE
- * kconfig option or it is detected at runtime.
- *
- * @param start_addr the pointer to start the multi-line flush
- * @param size the number of bytes that are to be flushed
- *
- * @return N/A
- */
-
-void sys_cache_flush(vaddr_t start_addr, size_t size)
+void arch_cache_flush(void *start_addr, size_t size, u32_t flags)
 {
+	ARG_UNUSED(flags);
```

Comment:
is anything else using `dcache_flush_mlines()`? could fold its body into this function if not

Reply:
Doesn't seem like any other function is using dcache_flush_mlines, will merge it with arch_cache_flush then. Thanks